Kernel/Syscalls/mmap.cpp

```diff
@@ -590,19 +590,36 @@ ErrorOr<FlatPtr> Process::sys$msyscall(Userspace<void*> address)
     return 0;
 }
 
-ErrorOr<FlatPtr> Process::sys$msync(Userspace<void*> address, size_t size, [[maybe_unused]] int flags)
+ErrorOr<FlatPtr> Process::sys$msync(Userspace<void*> address, size_t size, int flags)
 {
+    // Either MS_SYNC or MS_ASYNC shall be specified
+    // Only MS_SYNC, MS_ASYNC and MS_INVALIDATE are valid bits to be set in flag
+    if (((flags & MS_ASYNC) == MS_ASYNC) != ((flags & MS_SYNC) == MS_SYNC)
+        || (flags & (MS_SYNC | MS_ASYNC | MS_INVALIDATE)) != flags)
+        return EINVAL;
```

Comment:
These comments are redundant with the code you're writing, and can be removed by writing the code in a cleaner way.

```suggestion
    if ((flags & (MS_SYNC | MS_ASYNC | MS_INVALIDATE)) != flags)
        return EINVAL;
    bool is_async = (flags & MS_ASYNC) == MS_ASYNC;
    bool is_sync = (flags & MS_SYNC) == MS_SYNC;
    if (is_async && is_sync)
        return EINVAL;
```

Reply:
I'll go with that for now,
although I still think that having these helper variables just for a check is a bit overkill